semgrep_output_v1.atd

```diff
@@ -169,7 +169,15 @@ type core_match <python decorator="dataclass(frozen=True)"> = {
 
 (* TODO: try to make it as close as possible to 'cli_match_extra' below *)
 type core_match_extra <python decorator="dataclass(frozen=True)"> = {
-  ?message: string option; (* rule.message (?) *)
+  (* These fields generally come from the rule, but may be set here if they're
+   * being overriden for that particular finding. This would currently occur
+   * for rule with a validator for secrets, depending on what the valdiator
+   * might match, but could be expanded in the future. *)
+  ?message: string option;
+  ?metadata: raw_json option;
+  (* LATER: use a variant for severity *)
+  ?severity: string option;
```

Comment:
let's use core_severity here directly instead of those ugly string.
in cli_match_extra we're using a string because we have to, to remain backward compatible, but for new code we should do things in a clean way.

Reply:
You can add a (* since 1.44.0 *) comment before metadata and severity.

Reply:
Actually regarding core_severity, maybe just rename it "severity", extend it with Inventory and Experiment, so then
we can reuse this type instead of duplicating it everywhere.